src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```diff
@@ -77,21 +91,37 @@ public SwiftOutputStream(HttpURLConnection httpCon) throws IOException {
 
   @Override
   public void write(int b) throws IOException {
+    if (totalBytesWritten + INT_BYTE_SIZE >= maxSplitSize) {
+      splitFileUpload();
+      totalBytesWritten = 0;
+    }
     mOutputStream.write(b);
+    totalBytesWritten += INT_BYTE_SIZE;
   }
 
   @Override
   public void write(byte[] b, int off, int len) throws IOException {
+    if (totalBytesWritten + len >= maxSplitSize) {
+      splitFileUpload();
+      totalBytesWritten = 0;
+    }
     mOutputStream.write(b, off, len);
+    totalBytesWritten += len;
   }
 
   @Override
   public void write(byte[] b) throws IOException {
+    if (totalBytesWritten + b.length >= maxSplitSize) {
+      splitFileUpload();
+      totalBytesWritten = 0;
+    }
     mOutputStream.write(b);
+    totalBytesWritten += b.length;
   }
 
   @Override
   public void close() throws IOException {
+    LOG.info("{} bytes written", totalBytesWritten);
```

Comment:
@djalova I guess we don't want it at info level, otherwise it will be printed all the time. Can you remove it please? I recently try to reduce number of debug prints. If you like, you can make it at trace level